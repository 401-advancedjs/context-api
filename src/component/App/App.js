import React from 'react';
import CounterContext from '../context/counter-context.js';
import Counter from '../counter/counter.js'

function App() {
  return (
    <div className="parent">
    <CounterContext>
      <Counter />
    </CounterContext>
    </div>
  );
}

export default App;
