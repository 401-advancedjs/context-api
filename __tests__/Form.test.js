import React from 'react';
import Form from '../component/Form/Form.js';
import App from '../component/App/App.js';
import renderer from 'react-test-renderer';

describe('<Form />', () => {
  test('basic rendering', () => {
    const mountedForm = shallow(<Form />);
    expect(mountedForm.find('form')).toBeTruthy();
  });

  test('form input change', () => {
    const mountedApp = mount(<App />);
    const URL = mountedApp.find('input.URL');
    URL.simulate('change', {target: {value: 'Melissa'}});
    expect(mountedApp.state('URL')).toEqual('Melissa');

  });
  
  test('click on get', () => {
    const mountedApp = mount(<App />);
    const button = mountedApp.find('input.GET');
    button.simulate('change');
    expect(mountedApp.state('method')).toEqual('GET');
  });

  test('click on post', () => {
    const mountedApp = mount(<App />);
    const button = mountedApp.find('input.POST');
    button.simulate('change');
    expect(mountedApp.state('method')).toEqual('POST');
  });

  test('click on put', () => {
    const mountedApp = mount(<App />);
    const button = mountedApp.find('input.PUT');
    button.simulate('change');
    expect(mountedApp.state('method')).toEqual('PUT');
  });

  test('click on patch', () => {
    const mountedApp = mount(<App />);
    const button = mountedApp.find('input.PATCH');
    button.simulate('change');
    expect(mountedApp.state('method')).toEqual('PATCH');
  });

  test('click on delete', () => {
    const mountedApp = mount(<App />);
    const button = mountedApp.find('input.DELETE');
    button.simulate('change');
    expect(mountedApp.state('method')).toEqual('DELETE');
  });

  test('rendering follows the snapshot', () => {
    const snapshot = renderer.create(<Form />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});