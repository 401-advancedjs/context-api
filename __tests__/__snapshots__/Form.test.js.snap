// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> rendering follows the snapshot 1`] = `
<form>
  <div>
    <label>
      <input
        className="URL"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <div
    className="methodsDIV"
  >
    <label
      className="methods"
    >
      GET
      <input
        className="GET"
        name="methods"
        onChange={[Function]}
        type="radio"
        value="GET"
      />
    </label>
    <label
      className="methods"
    >
      POST
      <input
        className="POST"
        name="methods"
        onChange={[Function]}
        type="radio"
        value="POST"
      />
    </label>
    <label
      className="methods"
    >
      PUT
      <input
        className="PUT"
        name="methods"
        onChange={[Function]}
        type="radio"
        value="PUT"
      />
    </label>
    <label
      className="methods"
    >
      PATCH
      <input
        className="PATCH"
        name="methods"
        onChange={[Function]}
        type="radio"
        value="PATCH"
      />
    </label>
    <label
      className="methods"
    >
      DELETE
      <input
        className="DELETE"
        name="methods"
        onChange={[Function]}
        type="radio"
        value="DELETE"
      />
    </label>
    <button
      onClick={[Function]}
    >
      Go!
    </button>
  </div>
</form>
`;
